sql/template_tf.go

```diff
@@ -127,15 +127,18 @@ def validate_input_fn(batch_size):
 classifier.compile(optimizer=classifier.default_optimizer(),
     loss=classifier.default_loss(),
     metrics=["accuracy"])
-if hasattr(classifier, 'cluster_train_loop'):
-    classifier.cluster_train_loop(train_input_fn(BATCHSIZE))
+premade = hasattr(classifier, 'sqlflow_train_loop')
+if premade:
+    classifier.sqlflow_train_loop(train_input_fn(BATCHSIZE))
 else:
     classifier.fit(train_input_fn(BATCHSIZE),
         epochs=EPOCHS if EPOCHS else classifier.default_training_epochs(),
         verbose=VERBOSE)
 classifier.save_weights("{{.Save}}", save_format="h5")
 eval_result = classifier.evaluate(validate_input_fn(BATCHSIZE), verbose=VERBOSE)
-print("Training set accuracy: {accuracy:0.5f}".format(**{"accuracy": eval_result[1]}))
+
+if not premade:
```

Comment:
I think this condition can be the `label clause` is empty, no need to run the evaluation if there is no `label clause`:

``` golang
if "{{.Y.FeatureName}}" != "":
    eval_result = classifier.evaluate(validate_input_fn(BATCHSIZE), verbose=VERBOSE)
    print("Training set accuracy: {accuracy:0.5f}".format(**{"accuracy": eval_result[1]}))
```

Reply:
Good idea.

Reply:
It looks SQLFlow does not need to split the dataset into train&val while `Y.FeatureName` is empty.